lang/syncx/linkname.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 ByteDance Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncx
+
+import _ "unsafe"
+
+//go:noescape
+//go:linkname runtime_procPin runtime.procPin
+func runtime_procPin() int
+
+//go:noescape
+//go:linkname runtime_procUnpin runtime.procUnpin
+func runtime_procUnpin()
```

Comment:
Maybe we can reuse the same function in `https://github.com/bytedance/gopkg/blob/develop/internal/runtimex/ppin.go`

Reply:
ok sure